internal/deprecations_test.go

```diff
@@ -52,4 +57,777 @@ var _ = Describe("[internal] deprecations", func() {
+						Expect(containers[1].Resources.Limits).To(Equal(corev1.ResourceList{}))
+					})
+				})
+
+				It("should have the public IP source set to pod", func() {
+					Expect(spec.Services.PublicIPSource).NotTo(BeNil())
+					Expect(*spec.Services.PublicIPSource).To(Equal(fdbtypes.PublicIPSourcePod))
+				})
+
+				It("should have automatic replacements enabled", func() {
+					Expect(spec.AutomationOptions.Replacements.Enabled).NotTo(BeNil())
+					Expect(*spec.AutomationOptions.Replacements.Enabled).To(BeTrue())
+					Expect(spec.AutomationOptions.Replacements.FailureDetectionTimeSeconds).NotTo(BeNil())
+					Expect(*spec.AutomationOptions.Replacements.FailureDetectionTimeSeconds).To(Equal(1800))
+				})
+			})
+
+			Context("with the future defaults, changes only", func() {
+				JustBeforeEach(func() {
+					err := NormalizeClusterSpec(spec, DeprecationOptions{UseFutureDefaults: true, OnlyShowChanges: true})
+					Expect(err).NotTo(HaveOccurred())
+				})
+
+				It("should have default sidecar resource requirements", func() {
+					generalProcessConfig, present := spec.Processes[fdbtypes.ProcessClassGeneral]
+					Expect(present).To(BeTrue())
+					containers := generalProcessConfig.PodTemplate.Spec.Containers
+					Expect(len(containers)).To(Equal(1))
+					Expect(containers[0].Name).To(Equal("foundationdb-kubernetes-sidecar"))
+					Expect(containers[0].Resources.Requests).To(Equal(corev1.ResourceList{
+						"cpu":    resource.MustParse("100m"),
+						"memory": resource.MustParse("256Mi"),
+					}))
+					Expect(containers[0].Resources.Limits).To(Equal(corev1.ResourceList{
+						"cpu":    resource.MustParse("100m"),
+						"memory": resource.MustParse("256Mi"),
+					}))
+				})
+
+				It("should have no public IP source", func() {
+					Expect(spec.Services.PublicIPSource).To(BeNil())
+				})
+
+				It("should have automatic replacements enabled", func() {
+					Expect(spec.AutomationOptions.Replacements.Enabled).NotTo(BeNil())
+					Expect(*spec.AutomationOptions.Replacements.Enabled).To(BeTrue())
+				})
+
+				It("should have no configuration for other automatic replacement options", func() {
+					Expect(spec.AutomationOptions.Replacements.FailureDetectionTimeSeconds).To(BeNil())
+				})
+			})
+
+			Context("when applying future defaults on top of current explicit defaults", func() {
+				var originalSpec *fdbtypes.FoundationDBClusterSpec
+
+				BeforeEach(func() {
+					err = NormalizeClusterSpec(spec, DeprecationOptions{UseFutureDefaults: false, OnlyShowChanges: true})
+					Expect(err).NotTo(HaveOccurred())
+					originalSpec = spec.DeepCopy()
+				})
+
+				JustBeforeEach(func() {
+					err = NormalizeClusterSpec(spec, DeprecationOptions{UseFutureDefaults: true, OnlyShowChanges: true})
+					Expect(err).NotTo(HaveOccurred())
+				})
+
+				It("should be equal to the version with the old explicit defaults", func() {
+					Expect(reflect.DeepEqual(originalSpec, spec)).To(BeTrue())
```

Comment:
I believe a `semantic.DeepEqual` is the right choice here (like in https://github.com/FoundationDB/fdb-kubernetes-operator/blob/master/controllers/update_status.go#L312)